mesonbuild/mesonlib.py

```diff
@@ -277,6 +279,53 @@ def classify_unity_sources(compilers, sources):
             compsrclist[comp].append(src)
     return compsrclist
 
+class OrderedEnum(Enum):
+    """
+    An Enum which additionally offers homogeneous ordered comparison.
+    """
+    def __ge__(self, other):
+        if self.__class__ is other.__class__:
+            return self.value >= other.value
+        return NotImplemented
+
+    def __gt__(self, other):
+        if self.__class__ is other.__class__:
+            return self.value > other.value
+        return NotImplemented
+
+    def __le__(self, other):
+        if self.__class__ is other.__class__:
+            return self.value <= other.value
+        return NotImplemented
+
+    def __lt__(self, other):
+        if self.__class__ is other.__class__:
+            return self.value < other.value
+        return NotImplemented
+
+MachineChoice = OrderedEnum('MachineChoice', ['BUILD', 'HOST', 'TARGET'])
+
+class PerMachine:
+    def __init__(self, build, host, target):
+        self.build = build,
+        self.host = host
+        self.target = target
+
+    def __getitem__(self, machine: MachineChoice):
```

Comment:
We still have to support plain Python 3.5. That means no type annotations. Unfortunately.

Reply:
It doesn't enforce anything in 3.5 but the syntax (including name resolution) is supported. Try:
```python
>>> def a(b: int):
...     return b
...
>>> a('a')
'a'
>>> def a(b: iint):
...     return b
... 
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
NameError: name 'iint' is not defined
```
I find even without enforcement it is still useful for documentation's sake.